x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/QueryStringFunction.java

```diff
@@ -84,4 +87,20 @@ protected NodeInfo<? extends Expression> info() {
     public String getWriteableName() {
         return ENTRY.name;
     }
+
+    @Override
+    public EvalOperator.ExpressionEvaluator.Factory toEvaluator(
+        Function<Expression, EvalOperator.ExpressionEvaluator.Factory> toEvaluator
+    ) {
+        throw new UnsupportedOperationException("Evaluator for QueryStringQuery needs shard context");
+    }
+
+    @Override
+    public EvalOperator.ExpressionEvaluator.Factory toEvaluator(
+        Function<Expression, EvalOperator.ExpressionEvaluator.Factory> toEvaluator,
+        List<EsPhysicalOperationProviders.ShardContext> shardContexts
+    ) {
```

Comment:
I *think* we're better off changing the `Function` into an interface with both of these in it. But, again, proof of concept.